### ISSUE TITLE
Pointless suggestion for when pattern field is not mutable

### ISSUE THREAD

**Issue by sunjay** [rust-lang/rust#53937]
[Rust Playground](https://play.rust-lang.org/?gist=ca84599d211f96bcb072ada7d377231b&version=stable&mode=debug&edition=2015)

```rust
struct Foo {
    items: Vec<i32>,
}

fn main() {
    let Foo {items} = &Foo {items: vec![]};
    items.push(12);
}
```

This produces the error: 

```
error[E0596]: cannot borrow immutable borrowed content `*items` as mutable
 --> src/main.rs:7:5
  |
6 |     let Foo {items} = &Foo {items: vec![]};
  |              ----- consider changing this to `items`
7 |     items.push(12);
  |     ^^^^^ cannot borrow as mutable
```

The suggestion "consider changing this to `items`" is pointless in that it won't solve the problem.

**Comment by pnkfelix** [comment]
This is a dupe, probably of #51244. It is definitely fixed in beta and nightly. (i believe the fix was in PR #52242.)

**Comment by sunjay** [comment]
Thanks!
